actionpack/lib/action_dispatch/request/utils.rb

```diff
@@ -32,7 +32,7 @@ def self.check_param_encoding(params)
           unless params.valid_encoding?
             # Raise Rack::Utils::InvalidParameterError for consistency with Rack.
             # ActionDispatch::Request#GET will re-raise as a BadRequest error.
-            raise Rack::Utils::InvalidParameterError, "Non UTF-8 value: #{params}"
+            raise Rack::Utils::InvalidParameterError, "Enconding is invalid for paramter value: #{params.scrub}"
```

Comment:
I also changed the error message.
Because It doesn't mean that the parameter having an invalid encoding is the same as not been UTF8